src/scripts/main.js

```diff
@@ -1,3 +1,100 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const tbody = table.querySelector('tbody');
+const appendRowButton = document.querySelector('.append-row');
+const removeRowButton = document.querySelector('.remove-row');
+const appendColumnButton = document.querySelector('.append-column');
+const removeColumnButton = document.querySelector('.remove-column');
+
+appendRowButton.addEventListener('click', () => {
+  const columns = table.querySelector('tr').children.length;
+  const newRow = document.createElement('tr');
+
+  for (let i = 0; i < columns; i++) {
+    const newTd = document.createElement('td');
+
+    newRow.appendChild(newTd);
+  }
+
+  tbody.appendChild(newRow);
+
+  setButtonsState();
+});
+
+removeRowButton.addEventListener('click', () => {
+  tbody.removeChild(tbody.lastElementChild);
+
+  setButtonsState();
+});
+
+appendColumnButton.addEventListener('click', () => {
+  const rows = tbody.querySelectorAll('tr');
+  const rowLength = rows.length;
+
+  for (let i = 0; i < rowLength; i++) {
+    const newTd = document.createElement('td');
+
+    rows[i].appendChild(newTd);
+  }
+
+  setButtonsState();
+});
+
+removeColumnButton.addEventListener('click', () => {
+  const rows = tbody.querySelectorAll('tr');
+  const rowLength = rows.length;
+
+  for (let i = 0; i < rowLength; i++) {
+    rows[i].removeChild(rows[i].lastElementChild);
+  }
+
+  setButtonsState();
+});
+
+function setButtonsState() {
+  disableButtons();
+  enableButtons();
```

Comment:
The `setButtonsState` function calls both `disableButtons` and `enableButtons` functions. This might be redundant because these functions seem to perform opposite actions. Consider combining their logic into a single function to avoid unnecessary operations.